app/controllers/adminpanel_controller.rb

```diff
@@ -0,0 +1,47 @@
+# Administration Panel
+# frozen_string_literal: false
+
+# Administration user and administration panel
+class AdminpanelController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
the method seems to be `authenticate_user?` https://github.com/BC-MAR-21-ROR/proyecto-denuncia-ciudadana-proyecto-team03/pull/19/files#diff-766c34fd6533171eaf54300c153f89d6002c35c02cfc9c5b219251f85180ad07R11

Aside, following ruby conventions, the name with a `?` suffix will return a boolean and with `!` suffix will modify the object or will raise and exception when fail

https://gist.github.com/iangreenleaf/b206d09c587e8fc6399e#general-ruby-conventions
https://github.com/rubocop/ruby-style-guide#predicate-methods-suffix

Reply:
From what I have read the suffix `!` in that line is to check if the user is logged in.